basic.go

```diff
@@ -18,4 +19,7 @@ type BasicConnector interface {
 
 	// HTTPClient returns the underlying HTTP client. This is useful for proxy requests.
 	HTTPClient() *common.HTTPClient
+
+	// Provider returns the connector provider.
+	Provider() providers.Provider
```

Comment:
It's a bit redundant to have both the `Name` method and the `Provider` method (and it increases the overhead of creating new connectors. I think we should add Provider and remove `Name`, since `Provider().String()` is effectively equivalent to `Name()`